src/dialogs/chart_options/chart_options_model.pas

```diff
@@ -0,0 +1,135 @@
+unit chart_options_model;
+
+{$mode objfpc}{$H+}
+
+interface
+{
+ Valid options for all charts
+
+ !title
+ !footnote
+ !xtitle
+ !ytitle
+ !colors
+}
+uses
+  Classes, SysUtils, stat_dialog_contribution, chart_options, executor;
+
+type
+
+  TChartOptionsBox = (cbT, cbF, cbXT, cbYT, cbC, cbXmin, cbXmax, cbYmin, cbYmax);
+  
+  { TChartOptionsModel }
+
+  TChartOptionsModel = class(IStatDialogModel)
+  private
+    FExecutor: TExecutor;
+    FTitle:    UTF8String;
+    FFootnote: UTF8String;
+    FXTitle:   UTF8String;
+    FYTitle:   UTF8String;
+    FColors:   UTF8String;
+    FXMin, FXMax, FYMin, FYMax: UTF8String;
+    procedure SetTitle(AValue: UTF8String);
+    procedure SetFootnote(AValue: UTF8String);
+    procedure SetXTitle(AValue: UTF8String);
+    procedure SetYTitle(AValue: UTF8String);
+    procedure SetColors(AValue: UTF8String);
+
+  public
+    constructor Create(Executor: TExecutor);
+    function GenerateScript(): UTF8String;
+    function IsDefined(): boolean;
+  public
+    property Title: UTF8String read FTitle write SetTitle;
+    property Footnote: UTF8String read FFootnote write SetFootnote;
+    property XTitle: UTF8String read FXTitle write SetXTitle;
+    property YTitle: UTF8String read FYTitle write SetYTitle;
+    property Colors: UTF8String read FColors write SetColors;
+    property XMin: UTF8String read FXMin write FXMin;
```

Comment:
Either the view or the datamodel needs some sort of validate of the min/max fields. Since the input is directly written to the script the validation happens during runtime.
It would be nice if this input validation accounts for the datatype of variables on the axis.